internal/xds/translator/oidc.go

```diff
@@ -262,15 +278,15 @@ func createOAuth2Secrets(tCtx *types.ResourceVersionTable, routes []*ir.HTTPRout
 		}
 
 		clientSecret := buildOAuth2ClientSecret(route)
-		if err := tCtx.AddXdsResource(resourcev3.SecretType, clientSecret); err != nil {
+		if err := addXdsSecretIfNotExist(tCtx, clientSecret); err != nil {
```

Comment:
what happens when 2 different policies (attached to 2 different routes) access the same secret ?

Reply:
Good catch! 

Normally this shouldn't happen because different clients don't share the same secret. However, we can not assume users won't apply the same policy to different routes. I think we shouldn't return error in this case.

Reply:
That might happen, I choose to create a secret for each route and don't share secret across routes. I will add some comments so we (including me) won't get confused by the codes in the future.